pandas/core/reshape/reshape.py

```diff
@@ -115,6 +115,12 @@ def __init__(self, values, index, level=-1, value_columns=None,
 
         self.index = index.remove_unused_levels()
 
+        if isinstance(self.index, MultiIndex):
```

Comment:
I would rather not add API here. why don't you just do this check inside ``_get_level_number`` itself, and then also add to ``Index`` so you don't also have to add an instance check?

Reply:
Did you read my comment in the main thread? 
This code above is exactly how it was before. As I said, I also think this is better in het_level_number, but propose to do a follow-up PR that can be merged for 0.24.0, as it gives an additional API change (change in error type)

Reply:
then let's just wait till 0.24.0 to do this.

Reply:
It seems like a good fix for the regression, why not merge it? It's no different that reverting a previous commit.

Reply:
yes, it is how it was for a long time, and this is fixing a regression so it is the point to do it now and not wait for 0.24.0.

I can certainly do the PR for 0.24.0 to further clean this up, but that will involve an API change so I will not do that in this PR.